blog-posts/gatsby-ssr-caching.md

```diff
@@ -0,0 +1,98 @@
+is rendered on the server and delivered to the browser. Caching can be done on multiple levels in your architecture, in 
+this article we will focus on caching at the edge with a CDN.
+
+If you host your Gatsby page with Gatsby Cloud, you have a CDN out of the box. The CDN is powered by Fastly, but that's 
+just an implementation detail. Caching for your static (SSG) or deferred (DSG) pages is done automatically, so no need 
+to configure anything here. But if you want to do SSR, you need to take care of the correct caching behavior by yourself.
+
+> This post was inspired by the excellent video from [Remix on Youtube](https://www.youtube.com/watch?v=bfLFHp7Sbkg) 
+> If you want to deep dive to CDN caching with SSG and SSR, you should have a look! 
+
+## The Cache Control Header
+
+Caching in Gatsby Cloud can be controlled via the HTTP caching header `Cache-Control`. This header has multiple 
+[directives](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#directives), the most importantly:
+
+- `max-age`: The time in seconds for how long the page should be cached by the browser. Use this to determine how long the page should be cached  in the browser after the user fetched it. This cache is per user.
+- `s-maxage`: The time in seconds for how long the page should be cached by the CDN. If not set, the CDN will use `max-age`. Use this to determine how long a page can be cached globally. You may want to set the time lower for new and frequently changing content and higher for old or less frequently changing content.
+- `stale-while-revalidate` : The time in seconds for how long the page should be stored by the CDN and a new page should be requested in the background when a visitor hits it. Once the new page is requests, the stale page will be replaced by it. Use this to serve an old version of a page before the new page has been rendered. This is very powerful to keep your page fast while updating it in the background. If the content is not crucial, you can set this to a very high value.
+
+With those 3 directives, you can make your page as fast as with SSG or DSG. If the values are set correctly, only the 
+very first user after a deployment will have to wait for the rendering. Everyone else will see a lightning fast page.
+
+> This header can also be used on any other SSR framework and CDN. It's not limited to Gatsby or Gatsby Cloud.
+
+## Enable Caching
+
+Now let's dive into the code and set the `Cache-Control` header:
+
+```jsx
+// pages/ssr.jsx
+export default function Cache() {
+  return (
+      <h1>SSR: Server Side Rendering</h1>
+  )
+}
+
+export async function getServerData() {
+  return {
+    headers: {
+      'Cache-Control': 'public, max-age=10, s-maxage=60, stale-while-revalidate=240',
+    }
+  }
+}
+```
+
+Gatsby SSR pages export the `getServerData()` function, which you can use to do all kinds of stuff with it (like 
```

Comment:
Confusing, esp. with the sentence "This header can also be used on any other SSR framework and CDN. It's not limited to Gatsby or Gatsby Cloud." just before. 
Maybe add a sentence to "Now let's dive into the code and see how this can be done with Gatsby Cloud. The first step is to set the `Cache-Control`"...

Reply:
The sentence before is a callout, so I hope it doesn't confuse in the rendered post. Also changed the wording a bit.